src/lib/device-online-state/index.ts

```diff
@@ -0,0 +1,282 @@
+import * as Bluebird from 'bluebird';
+declare global {
+	export interface Promise<T> extends Bluebird<T> {}
+}
+import { EventEmitter } from 'events';
+import { DEFAULT_SUPERVISOR_POLL_INTERVAL } from '../env-vars';
+import { noop } from 'lodash';
+import { resinApi, root } from '../../platform';
+import { captureException } from '../../platform/errors';
+import {
+	createPromisifedRedisClient,
+	PromisifedRedisClient,
+} from '../redis-promise';
+import * as RedisSMQ from 'rsmq';
+import { REDIS_HOST, REDIS_PORT, API_HEARTBEAT_STATE_TIMEOUT } from '../config';
+
+export class PollIntervalNotFoundError extends Error {}
+export const getPollIntervalForDevice = (uuid: string): Bluebird<number> => {
+	return resinApi
+		.get({
+			resource: 'device_config_variable',
+			passthrough: {
+				req: root,
+			},
+			options: {
+				$select: 'value',
+				$expand: {
+					device: {
+						$filter: { uuid },
+					},
+				},
+				$filter: {
+					name: 'RESIN_SUPERVISOR_POLL_INTERVAL',
+				},
+			},
+		})
+		.then(([pollInterval]: Array<{ value: number }>) => {
+			if (!pollInterval) {
+				throw new PollIntervalNotFoundError();
```

Comment:
Why introduce this sort of indirection here? I mean you throw it in here and just catch it and handle it in the catch handler a few lines below.